Core/include/Acts/Seeding/detail/FastStrawLineFitter.hpp

```diff
@@ -0,0 +1,218 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/EventData/CompositeSpacePoint.hpp"
+#include "Acts/Seeding/detail/CompSpacePointAuxiliaries.hpp"
+#include "Acts/Utilities/CalibrationContext.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <memory>
+#include <vector>
+
+namespace Acts::Experimental::detail {
+
+///  @brief The FastStrawLineFitter fits a straight line to a set of straw measurements
+///         The space points  passed to the fitter need to be all straw space
+///         points and their straw wires need to be approximately parallel to
+///         each other. Under this assumption, each straw measurements can be
+///         projected onto the plane which is pependicular to its straw wire and
+///         the residual to the k-th straw is written as
+///
+///                 R_{k} = T_{z,k} * sin \theta - (T_{y,k} -  y_{0})* cos \theta - sign_{k} * r_{k}
+///
+///         where, \theta and y_{0} are the two line parameters to fit, T_{z,k} and T_{y,k} are
+///         the coordinates of the straw-tube in the plane, sign_{k} fixes the
+///         left-right ambiguity, and r_{k} is the drift radius of the
+///         measurement. The z & y coordinates are given by the dot product of
+///         the straw's local position with its planeNormal & toNextSensor
+///         vector, respectively. Assuming that the drift radius does not need
+///         to be calibrated during the fit, the entire problem reduces to the
+///         minimization of a polynomial having the form
+///
+///                   A * cos(\theta)  + B * sin(\theta)  + C * cos(2\theta) + D * sin(2\theta)
+///         where A, B, C, D are fit constants depending on the configuration of
+///         the tubes to fit. They are documented in
+///         https://gitlab.cern.ch/atlas-nextgen/work-package-2.5/analyticalsegment
+///
+class FastStrawLineFitter {
+ public:
+  /// @brief abrivation of the residual indices to fetch the proper covariance
+  using ResidualIdx = CompSpacePointAuxiliaries::ResidualIdx;
+  /// @brief Vector type
+  using Vector = CompSpacePointAuxiliaries::Vector;
+  /// @brief Configuration object
+  struct Config {
+    /// @brief Number of maximum iterations
+    std::uint32_t maxIter{10};
```

Comment:
```suggestion
    std::size_t maxIter{10};
```
Also for the other `uint`-cases